react-calculator/src/hooks/useCalculate.tsx

```diff
@@ -0,0 +1,96 @@
+import { useState } from "react";
+import { OperationType } from "../components/Opertaions";
+
+const useCalculate = () => {
+  const [firstNumber, setFirstNumber] = useState("");
+  const [secondNumber, setSecondNumber] = useState("");
+  const [resultNumber, setResultNumber] = useState(0);
+  const [operation, setOperation] = useState<OperationType>();
+  const [done, setDone] = useState(false);
+
+  const state = {
+    firstNumber,
+    secondNumber,
+    resultNumber,
+    operation,
+    done,
+  };
+
+  const firstNumberHandler = (num: number) => {
+    if (firstNumber.length >= 3) return;
+    const number = firstNumber + num;
+    setFirstNumber(number);
+  };
+  const secondNumberHandler = (num: number) => {
+    if (secondNumber.length >= 3) return;
+    const number = secondNumber + num;
+    setSecondNumber(number);
+  };
+
+  const numHandler = (num: number) => {
+    if (!operation) {
+      firstNumberHandler(num);
+      return;
+    }
```

Comment:
이 코드가 조금 더 깔끔해 보이네요☺

```suggestion
    if (!operation) return firstNumberHandler(num);
```

Reply:
코드 양이 줄어보이고 훨씬 깔끔해지고 좋은 것 같습니다.

그런데, 코드를 리뷰하는 입장에서 firstNumberHandler 함수가 리턴타입이 있는지 모른다고 한다면, 한번 확인해야하는 공수가 들것 같은데 클린코드 입장에서 어떤 부분이 더 옳은지가 궁금합니다.

Reply:
오... 어차피 타입스크립트라면 리턴 타입이 추론되지 않을까요? void가 나오지 않을까 싶습니다.